File: src/main/java/org/smudge/petstore/PersonService.java
package org.smudge.petstore;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class PersonService {

    @Autowired
    private PersonRepository repo;

    public Person getPerson(int id) {
        return repo.findById(id).get().toBusinessObject();
    }

    public Person createPerson(String name) {
        PersonEntity newPerson = new PersonEntity(0, name);
        repo.save(newPerson);
        return newPerson.toBusinessObject();
    }

    public Person changeName(int id, String newName) {
        Person person = getPerson(id);
        person.setName(newName);
        return person;
    }
}

File: src/main/java/org/smudge/petstore/PersonService.kt
package org.smudge.petstore

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class PersonService(
        @Autowired
        private val repo: PersonRepository
) {

    fun getPerson(id: Int): Person {
        return repo.findById(id).orElse(null)?.toBusinessObject()
                ?: throw Exception("No person found with id $id")
    }

    fun createPerson(name: String): Person {
        val newPerson = PersonEntity(0, name)
        repo.save(newPerson)
        return newPerson.toBusinessObject()
    }

    fun changeName(id: Int, newName: String): Person {
        val person = repo.findById(id).get()
        person.name = newName
        repo.save(person)
        return person.toBusinessObject()
    }
}